client/src/components/addCenterComponent.js

```diff
@@ -0,0 +1,234 @@
+import React from 'react';
+import { ModalHeader } from './modalHeader';
+import { FormGroup } from './formGroup';
+import centerNameIcon from '../resources/images/glyphicons-21-home.png';
+import centerImageIcon from '../resources/images/glyphicons-139-picture.png';
+import addressIcon from '../resources/images/glyphicons-243-map-marker.png';
+import cityIcon from '../resources/images/glyphicons-740-cadastral-map.png';
+import capacityIcon from '../resources/images/glyphicons-44-group.png';
+import amountIcon from '../resources/images/glyphicons-228-usd.png';
+import { Option } from './selectOption';
+import axios from 'axios';
+import { logout, apiLink } from '../reusables';
+import { connect } from 'react-redux';
+import { setCenterDetails } from '../actions/centerActions';
+
+const inputAttrs = (inputType, inputName, placeholder, className, ref, required) => ({
+  inputType,
+  inputName,
+  placeholder,
+  className,
+  ref,
+  required,
+});
+
+const mapDispatchToProps = dispatch => ({
+  setCenterDetails: center => dispatch(setCenterDetails(center)),
+  // setCenterDefaults: data => dispatch(setCenterDefaults(data)),
+});
+const mapStateToProps = state => ({
+  modalTitle: state.page.modalTitle,
+  required: state.page.required,
+  centerDefaults: state.page.centerDefaults,
+});
+let centerId;
+
+class AddCenterComponent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.folder = apiLink === 'http://localhost:8080' ? 'dev/centers' : 'prod/centers';
+  }
+  handleSubmit(event) {
+    event.preventDefault();
+    const name = this.name.value;
+    const address = this.address.value;
+    const location = this.location.value;
+    const capacity = this.capacity.value;
+    const price = this.price.value;
+    const availability = this.availability.value;
+    const token = JSON.parse(localStorage.token).value;
```

Comment:
I don't think these constants above are necessary. Most of them are only used once after the definition.

Reply:
Thanks, will refactor asap